src/Helpers/CrumbHelper.cs

```diff
@@ -1,87 +1,167 @@
         {
-            _instance = null;
+            await Instance.SetCrumbAsync();
         }
 
-        public async Task SetCrumbAsync()
-        {
-            var client = GetHttpClient();
-            var loginResponse = await client.GetAsync("https://login.yahoo.com/");            
+        return Instance;
+    }
+
+    private static HttpClientHandler GetClientHandler()
+    {
+        return YahooClient.IsThrottled
+            ? new DownloadThrottleQueueHandler(40, TimeSpan.FromMinutes(1),4) //40 calls in a minute, no more than 4 simultaneously
+            : new HttpClientHandler();
+    }
+
+    #endregion
+
+    #region Public Methods
 
-            if (loginResponse.IsSuccessStatusCode)
+    public async Task SetCrumbAsync()
+    {
+        var client = GetHttpClient();
+        var loginResponse = await client.GetAsync("https://login.yahoo.com/");
+
+        if (loginResponse.IsSuccessStatusCode)
+        {
+            var login = await loginResponse.Content.ReadAsStringAsync();
+            if (loginResponse.Headers.TryGetValues("Set-Cookie", out var setCookie))
             {
-                var login = await loginResponse.Content.ReadAsStringAsync();
-                if (loginResponse.Headers.TryGetValues(name: "Set-Cookie", out IEnumerable<string>? setCookie))
-                {
-                    cookies = new List<string>(setCookie.Where(c => c.ToLower().IndexOf("domain=.yahoo.com") > 0));                   
-                    var crumbResponse = await client.GetAsync("https://query1.finance.yahoo.com/v1/test/getcrumb");
+                cookies = new List<string>(setCookie.Where(c => c.ToLower().IndexOf("domain=.yahoo.com") > 0));
+                var crumbResponse = await client.GetAsync("https://query1.finance.yahoo.com/v1/test/getcrumb");
 
-                    if (crumbResponse.IsSuccessStatusCode)
-                    {
-                        Crumb = await crumbResponse.Content.ReadAsStringAsync();
-                    }
+                if (crumbResponse.IsSuccessStatusCode)
+                {
+                    Crumb = await crumbResponse.Content.ReadAsStringAsync();
                 }
             }
+        }
 
-            if (string.IsNullOrEmpty(Crumb))
-            {
-                throw new Exception("Failed to get crumb");
-            }
+        if (string.IsNullOrEmpty(Crumb))
+        {
+            throw new Exception("Failed to get crumb");
         }
+    }
+
+    #endregion
+
+    #region Internal Methods
+
+    internal void Destroy()
+    {
+        _instance = null;
+    }
+
+    #endregion
+}
 
-        public static HttpClient GetHttpClient()
+internal class DownloadThrottleQueueHandler : HttpClientHandler
+{
+    #region Fields
+
+    private readonly TimeSpan _maxPeriod;
+    private readonly SemaphoreSlim _throttleLoad, _throttleRate;
+
+    #endregion
+
+    public DownloadThrottleQueueHandler(int maxPerPeriod, TimeSpan maxPeriod, int maxParallel = -1)
```

Comment:
Did you do testing with the 40 per minute limit to confirm this is Yahoo's hard limit?

Reply:
I did not test the 40 per minute, but used the number based upon some comments in the community - typically 60 per minute is quoted, but there is no hard evidence to support.